oracle-plugin/docs/Oracle-batchsource.md

```diff
@@ -53,6 +53,37 @@ will be passed to the JDBC driver as connection arguments for JDBC drivers that
 **Default Row Prefetch:** The default number of rows to prefetch from the server.
 
 
+Data Types Mapping
+----------
+
+    | Oracle Data Type               | CDAP Schema Data Type | Comment                |
+    | ------------------------------ | --------------------- | ---------------------- |
+    | VARCHAR2                       | string                |                        |
+    | NVARCHAR2                      | string                |                        |
+    | VARCHAR                        | string                |                        |
+    | NUMBER                         | decimal               |                        |
+    | FLOAT                          | double                |                        |
+    | LONG                           | string                |                        |
+    | DATE                           | timestamp             |                        |
+    | BINARY_FLOAT                   | float                 |                        |
+    | BINARY_DOUBLE                  | double                |                        |
+    | TIMESTAMP                      | timestamp             |                        |
+    | TIMESTAMP WITH TIME ZONE       | timestamp             | Converted to UTC time  |
```

Comment:
In https://wiki.cask.co/display/CE/Oracle+database+plugin, I mentioned that we should read these as strings until there is a way to represent this natively in the CDAP Schema. That way no data is lost (users can always convert from a string to a timestamp using a wrangler transform after the source).

Was this just an issue for Oracle? I imagine the other DBs have the same issue?

Reply:
Mapping of `TIMESTAMP WITH TIME ZONE` changed to `Types.STRING`. There was the same issue with 'DATETIMEOFFSET' MS SQL Server data type, but we mapped it to the `Types.STRING`.